File: app/src/main/java/getyourcasts/jd/com/getyourcasts/repository/local/SqliteHelper.java
package getyourcasts.jd.com.getyourcasts.repository.local;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import org.jetbrains.anko.db.ManagedSQLiteOpenHelper;

/**
 * Created by chuondao on 8/8/17.
 */

public class SqliteHelper extends SQLiteOpenHelper {


    static final String DB_NAME = "PODCAST_DB";
    static final int VERSION = 1;
    private static SqliteHelper instance = null;

    public SqliteHelper(Context ctx, String name, SQLiteDatabase.CursorFactory factory, int version) {
        super(ctx, name, factory, version);
    }

    public static synchronized SqliteHelper getInstance(Context ctx) {
        if (instance == null) {
            instance = new SqliteHelper(ctx, DB_NAME, null, VERSION);
        }
        return instance;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        // create two tables
        db.execSQL(getCreateEpisodeTableCmd());
        db.execSQL(getCreatePodcastTableCmd());
    }


    private String getCreatePodcastTableCmd() {

        return "CREATE TABLE " + Contract.PodcastTable.TABLE_NAME + " ("
                +Contract.PodcastTable.ID + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                +Contract.PodcastTable.PODCAST_NAME + " TEXT, "
                +Contract.PodcastTable.LAST_UPDATE + " TEXT, "
                +Contract.PodcastTable.LAST_UPDATE + " TEXT, "
                +Contract.PodcastTable.IMG_ONLINE_PATH + " TEXT, "
                +Contract.PodcastTable.FEED_URL + " TEXT, "
                +Contract.PodcastTable.TRACK_COUNT + " INTEGER, "
                +Contract.PodcastTable.ARTIST_NAME + " TEXT, "
                +Contract.PodcastTable.IMG_LOCAL_PATH + " TEXT, "
                +Contract.PodcastTable.RELEASE_DATE + " TEXT, "
                +Contract.PodcastTable.DESCRIPTION + " TEXT, "
                +Contract.PodcastTable.UNIQUE_ID + " TEXT UNIQUE, ";

    }

    private String getCreateEpisodeTableCmd() {

        return "CREATE TABLE " + Contract.EpisodeTable.TABLE_NAME + " ("
                +Contract.EpisodeTable.ID + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                +Contract.EpisodeTable.EPISODE_NAME + " TEXT, "
                +Contract.EpisodeTable.PODCAST_ID + " TEXT, "
                +Contract.EpisodeTable.UNIQUE_ID + " TEXT UNIQUE, "
                +Contract.EpisodeTable.DATE_DOWNLOADED + " TEXT, "
                +Contract.EpisodeTable.DATE_RELEASED + " TEXT, "
                +Contract.EpisodeTable.PROGRESS + " TEXT, "
                +Contract.EpisodeTable.FILE_SIZE + " TEXT, "
                +Contract.EpisodeTable.LOCAL_URL + " TEXT, "
                +Contract.EpisodeTable.MEDIA_TYPE + " TEXT, "
                +Contract.EpisodeTable.DESCRIPTION + " TEXT, "
                +Contract.EpisodeTable.FAVORITE + " TEXT, "
                +Contract.EpisodeTable.FETCH_URL + " TEXT, "
                +Contract.EpisodeTable.DOWNLOADED + " TEXT, ";
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(" DROP TABLE IF EXISTS " + Contract.EpisodeTable.TABLE_NAME);
        db.execSQL(" DROP TABLE IF EXISTS " + Contract.PodcastTable.TABLE_NAME);
        onCreate(db);
    }
}

File: app/src/main/java/getyourcasts/jd/com/getyourcasts/repository/local/SqliteHelper.kt
package getyourcasts.jd.com.getyourcasts.repository.local

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import org.jetbrains.anko.db.*

/**
 * Created by chuondao on 7/21/17.
 */
class PodCastSqliteHelper (ctx: Context): ManagedSQLiteOpenHelper(ctx, PodCastSqliteHelper.DB_NAME, null,
        PodCastSqliteHelper.VERSION ) {


    companion object {
        val DB_NAME = "PODCAST_DB"
        val VERSION = 1
        private var instance: PodCastSqliteHelper? = null

        @Synchronized
        fun getInstance(ctx: Context): PodCastSqliteHelper {
            if (instance == null) {
                instance = PodCastSqliteHelper(ctx.getApplicationContext())
            }
            return instance!!
        }
    }

    override fun onCreate(db: SQLiteDatabase?) {

        if (db != null){
            // create PODCAST TABLE
            db.createTable(
                    PodcastsTable.NAME, true,
                    PodcastsTable.ID to INTEGER + PRIMARY_KEY,
                    PodcastsTable.UNIQUE_ID to TEXT + UNIQUE,
                    PodcastsTable.PODCAST_NAME to TEXT,
                    PodcastsTable.LAST_UPDATE to TEXT,
                    PodcastsTable.IMG_ONLINE_PATH to TEXT,
                    PodcastsTable.FEED_URL to TEXT ,
                    PodcastsTable.TRACK_COUNT to INTEGER,
                    PodcastsTable.ARTIST_NAME to TEXT,
                    PodcastsTable.IMG_LOCAL_PATH to TEXT,
                    PodcastsTable.RELEASE_DATE to TEXT,
                    PodcastsTable.DESCRIPTION to TEXT
            )

            // create EPISODE TABLE
            db.createTable(
                    EpisodeTable.NAME, true,
                    EpisodeTable.ID to INTEGER + PRIMARY_KEY,
                    EpisodeTable.EPISODE_NAME to TEXT,
                    EpisodeTable.PODCAST_ID to TEXT,
                    EpisodeTable.UNIQUE_ID to TEXT,
                    EpisodeTable.DATE_DOWNLOADED  to TEXT,
                    EpisodeTable.DATE_RELEASED to TEXT,
                    EpisodeTable.PROGRESS to INTEGER,
                    EpisodeTable.FILE_SIZE to TEXT,
                    EpisodeTable.FETCH_URL to TEXT,
                    EpisodeTable.LOCAL_URL to TEXT,
                    EpisodeTable.MEDIA_TYPE to TEXT,
                    EpisodeTable.DESCRIPTION to TEXT,
                    EpisodeTable.FAVORITE to INTEGER,
                    EpisodeTable.DOWNLOADED to INTEGER
            )

        }

        // create EPISODE TABLE
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        if (db != null){
            db.dropTable(PodcastsTable.NAME, true)
            db.dropTable(EpisodeTable.NAME, true)
            onCreate(db)
        }
    }

}

val Context.database: PodCastSqliteHelper
    get() = PodCastSqliteHelper.getInstance(applicationContext)
